python/cudf/cudf/core/index.py

```diff
@@ -1212,37 +1161,63 @@ def get_loc(self, key, method=None, tolerance=None):
                 "is specified."
             )
 
-        key_as_table = cudf.core.frame.Frame(
-            {"None": as_column(key, length=1)}
+        needle_table = cudf.DataFrame(
+            {"None": as_column(target), "order": arange(0, len(target))}
+        )
+        haystack_table = cudf.DataFrame(
+            {"None": self._column, "order": arange(0, len(self))}
         )
+        merged_table = haystack_table.merge(
+            needle_table, on="None", how="outer"
+        )
+        result_series = (
+            merged_table.sort_values(by="order_y")
+            .head(len(target))["order_x"]
+            .reset_index(drop=True)
+        )
+        if method is None:
+            result_series = result_series.fillna(-1)
+        elif method in {"ffill", "bfill", "pad", "backfill"}:
+            result_series = _get_indexer_basic(
+                index=self,
+                positions=result_series,
+                method=method,
+                target_col=needle_table["None"],
+                tolerance=tolerance,
+            )
+        elif method == "nearest":
+            result_series = _get_nearest_indexer(
+                index=self,
+                positions=result_series,
+                target_col=needle_table["None"],
+                tolerance=tolerance,
+            )
+        else:
+            raise ValueError(
+                f"{method=} is unsupported, only supported values are: "
+                "{['ffill'/'pad', 'bfill'/'backfill', 'nearest', None]}"
+            )
+
+        return result_series.to_cupy()
+
+    @_cudf_nvtx_annotate
+    def get_loc(self, key):
+        if is_scalar(key):
+            target = [key]
+        else:
+            target = key
```

Comment:
I _think_ this is wrong somehow? `pandas` raises `InvalidIndexError` in `get_loc` if the key is not scalar.

Reply:
Good catch, fixed this issue.